### PR TITLE
ANNULUS-243 Update RPC Calls

### PR DESCRIPTION
**Pull request by berriose**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] I have added screenshots

### CODE REVIEW

**Review by KalervoHyyppa — CHANGES_REQUESTED** [review]
Putting this here to give me some time to review

File: lib/blocks/providers/block_provider.dart [hunk 1]
```diff
@@ -1,76 +1,126 @@
...
+
+  final config = ref.read(configProvider.future);
+  final presentConfig = await config;
+
+  var blockRes = await genusClient.getBlockByDepth(depth: depth);
+
+  return Block(
+    header: getChainId(blockRes.block.header.headerId.value),
+    epoch: blockRes.block.header.slot.toInt() ~/ presentConfig.config.epochLength.toInt(),
+    size: 5432.2,
```
Comment (KalervoHyyppa) [inline lib/blocks/providers/block_provider.dart:22]: Where does the block size come from?

File: lib/blocks/providers/block_provider.dart [hunk 2]
```diff
@@ -1,76 +1,126 @@
...
+
+  final config = ref.read(configProvider.future);
+  final presentConfig = await config;
+
+  var blockRes = await genusClient.getBlockByHeight(height: height);
+
+  return Block(
+    header: getChainId(blockRes.block.header.headerId.value),
+    epoch: blockRes.block.header.slot.toInt() ~/ presentConfig.config.epochLength.toInt(),
+    size: 5432.2,
```
Comment (KalervoHyyppa) [inline lib/blocks/providers/block_provider.dart:42]: Same for size here

File: lib/blocks/providers/block_provider.dart (hunk 1 above)
Comment (berriose) [reply lib/blocks/providers/block_provider.dart:22]: It's a placeholder. I didn't see a way to get the size in the methods.

**Review by KalervoHyyppa — CHANGES_REQUESTED** [review]
Block provider looks good

File: lib/transactions/providers/transactions_provider.dart [hunk 3]
```diff
@@ -1,22 +1,130 @@
...
+  final List<Transaction> transactions = [];
+  //get most recent block
+  final selectedChain = ref.read(selectedChainProvider.notifier).state;
+  final genusClient = ref.read(genusProvider(selectedChain));
+
+  var blockRes = await genusClient.getBlockByDepth(depth: depth);
+  final config = ref.read(configProvider.future);
+  final presentConfig = await config;
+
+  if (blockRes.block.fullBody.hasField(1)) {
```
[the remark this answers is not in the document]
Comment (berriose) [reply lib/transactions/providers/transactions_provider.dart:69]: I'll add a comment there. The fullbody is empty if there are no transactions. This checks for transactions.

**Review by KalervoHyyppa — APPROVED** [review]
Looks good!

**Pull request merged**